src/pretix/plugins/sendmail/models.py

```diff
@@ -0,0 +1,400 @@
+        """
+        return sorted(set(r.email.strip().lower() for r in self.recipients.all() if r.email))
+    
+    def populate_to_users(self, save=True):
+        """
+        Resolves recipients and populates to_users with metadata.
+        """
+        from collections import defaultdict
+
+        filters = getattr(self, 'filters_data', None)
+        if not filters:
+            return
+
+        recipients_mode = filters.recipients or "orders"
+        orders_qs = Order.objects.filter(
+            pk__in=filters.orders,
+            event=self.event
+        ).prefetch_related('positions__item', 'positions__addons', 'positions__checkins')
+
+        recipients = defaultdict(lambda: {
+            "orders": set(),
+            "positions": set(),
+            "items": set()
+        })
+
+        for order in orders_qs:
+            order_fallback_needed = False
+            attendee_found = False
+
+            for pos in order.positions.all():
+                if pos.attendee_email:
+                    attendee_found = True
+                    email = pos.attendee_email.strip().lower()
+                    recipients[email]["orders"].add(str(order.pk))
+                    recipients[email]["positions"].add(str(pos.pk))
+                    recipients[email]["items"].add(str(pos.item.pk))
+                else:
+                    # No attendee email; maybe fallback later
+                    order_fallback_needed = True
+
+            # Fallback to order email if needed
+            if (
+                order_fallback_needed and
+                not attendee_found and
+                recipients_mode == "attendees" and
+                order.email
+            ):
+                email = order.email.strip().lower()
+                recipients[email]["orders"].add(str(order.pk))
+                item_ids = order.positions.values_list('item__pk', flat=True)
+                recipients[email]["items"].update(str(iid) for iid in item_ids)
+
+            # Explicit inclusion of orders
+            if recipients_mode in ("both", "orders") and order.email:
+                email = order.email.strip().lower()
+                recipients[email]["orders"].add(str(order.pk))
+                item_ids = order.positions.values_list('item__pk', flat=True)
+                recipients[email]["items"].update(str(iid) for iid in item_ids)
+
+        # Clear and insert fresh records
+        self.recipients.all().delete()
+
+        objs = [
+            QueuedMailToUsers(
+                mail=self,
+                email=email,
+                orders=list(data["orders"]),
+                positions=list(data["positions"]),
+                items=list(data["items"]),
+                sent=False,
+                error=None,
+            )
+            for email, data in recipients.items()
+        ]
+
+        QueuedMailToUsers.objects.bulk_create(objs)
+        if save:
+            self.save()
+
+
+class QueuedMailToUsers(models.Model):
```

Comment:
Why name the model with plural (and the comment below is "epresents a single recipient"?